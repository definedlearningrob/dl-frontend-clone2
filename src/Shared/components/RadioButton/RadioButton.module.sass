@import '@shared/stylesheets/config.sass'

$inputSize: 20px

.labelWrapper
  top: 1px
  position: relative
  line-height: $inputSize

.radioLabel
  +flex-container(row)
  cursor: pointer
  position: relative
  color: $color-font-primary

  input[type="radio"]:focus
    outline-color: $color-primary-500

  input[type="radio"]
    margin-right: $m-xs
    appearance: none
    font: inherit
    color: currentColor
    width: $inputSize
    height: $inputSize
    border: $m-xxxs solid $color-neutral-300
    border-radius: 50%
    display: grid
    place-content: center

  input[type="radio"]:not(:disabled):hover
    border-color: $color-neutral-400

  input[type="radio"]:before
    content: ""
    width: $inputSize
    height: $inputSize
    border-radius: 50%
    transform: scale(0)
    box-shadow: inset 1em 1em transparent
    border:  4px solid $color-primary-500
    transition: all .2s ease-in

  input[type="radio"]:checked:before
    transform: scale(1)

  input[type="radio"]:disabled
    background-color: $color-neutral-300

  input[type="radio"]:disabled:checked:before
    border-color: $color-neutral-300
    background-color: $color-neutral-200
