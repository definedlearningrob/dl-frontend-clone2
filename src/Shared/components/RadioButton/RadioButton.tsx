import React, { InputHTMLAttributes } from 'react';
import cx from 'classnames';

import styles from './RadioButton.module.sass';

export type RadioButtonProps = InputHTMLAttributes<HTMLInputElement> & {
  id?: string;
  textClassName?: string;
};

export const RadioButton = ({
  id,
  name,
  value,
  children,
  onChange,
  checked,
  className,
  textClassName,
  ...props
}: RadioButtonProps) => {
  const radioWrapper = cx(styles.radioLabel, className, { '!cursor-not-allowed': props.disabled });

  return (
    <label className={radioWrapper}>
      <input
        {...props}
        checked={checked}
        className={styles.radioInput}
        id={id}
        name={name}
        type='radio'
        value={value}
        onChange={onChange}
      />
      <div className={styles.radioSpan}>
        <span className={cx(styles.labelWrapper, textClassName)}>{children}</span>
      </div>
    </label>
  );
};
