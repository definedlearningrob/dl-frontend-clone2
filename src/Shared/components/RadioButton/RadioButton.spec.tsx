import { fireEvent, render, screen } from '@testing-library/react';

import { RadioButton } from '@shared/components/RadioButton/RadioButton';

describe('RadioButton', () => {
  it('should render correctly with passed children', () => {
    const { container } = render(
      <RadioButton name='radio name' value='radio value' onChange={jest.fn()}>
        Radio label
      </RadioButton>
    );

    expect(screen.getByText('Radio label')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should call onChange when clicking radio button', () => {
    const onChangeMock = jest.fn();
    render(
      <RadioButton name='radio name' value='radio value' onChange={onChangeMock}>
        Radio label
      </RadioButton>
    );

    fireEvent.click(screen.getByRole('radio'));

    expect(onChangeMock).toHaveBeenCalled();
  });
});
