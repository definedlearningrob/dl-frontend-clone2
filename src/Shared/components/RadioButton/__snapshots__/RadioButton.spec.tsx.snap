// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton should render correctly with passed children 1`] = `
<div>
  <label
    class="radioLabel"
  >
    <input
      class="radioInput"
      name="radio name"
      type="radio"
      value="radio value"
    />
    <div
      class="radioSpan"
    >
      <span
        class="labelWrapper"
      >
        Radio label
      </span>
    </div>
  </label>
</div>
`;
